transitions/core.py

```diff
@@ -359,16 +356,17 @@ def __init__(self, model=None, states=None, initial=None, transitions=None,
         self._transition_queue = deque()
         self.models = []
 
-        if initial is None:
-            self.add_states('initial')
+        if model is None and add_self:  # TODO(pbovbel) reconsider API for next major release
+            model = self
+
```

Comment:
That might be getting too clever:

```
initial=False - no initial state, Error if models provided or add_self=True in __init__

initial=None - default, create 'initial' state for machine

initial=something - must be a valid state in machine
```

Torn whether implementing that is worthwhile rather than just enforcing that SOME initial state is provided. However, that would allow initialising a machine with **_nothing_**, and then dynamically adding states, transitions and models piecemeal.